### PR TITLE
Remove deprecated text from probes task

### PR DESCRIPTION
**Pull request by tengqm**
There are some text useful before v1.23 release, but no longer supported. 
Now we are moving on to v1.28 as the lastest version, these contents become obsolete and confusing. We'd better remove them because docs related to pre-1.24 releases are no longer supported.

### CODE REVIEW

File: content/en/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md [hunk 1]
```diff
@@ -518,22 +496,13 @@ to resolve it.
...
-was set.
-
 In 1.25 and above, users can specify a probe-level `terminationGracePeriodSeconds`
 as part of the probe specification. When both a pod- and probe-level 
 `terminationGracePeriodSeconds` are set, the kubelet will use the probe-level value.
 
-{{< note >}}
-Beginning in Kubernetes 1.25, the `ProbeTerminationGracePeriod` feature is enabled
-by default. For users choosing to disable this feature, please note the following:
+For users choosing to disable this feature, please note the following:
```
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md:503]: Can you disable it?

Comment (tengqm) [reply content/en/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md]: Double checked the source, it looks like the gate is "locked-to-default", which means the gate is recognized but it cannot be disabled.
Revised the text accordingly.

**Review by sftim — COMMENTED** [review]
LGTM for docs.

Tech review optional IMO.
/lgtm

File: content/en/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md [hunk 2]
```diff
@@ -274,19 +274,14 @@ After 15 seconds, view Pod events to verify that the liveness check has not fail
...
-The built-in gRPC probe's behavior is similar to the one implemented by grpc-health-probe.
-When migrating from grpc-health-probe to built-in probes, remember the following differences:
-
-- Built-in probes run against the pod IP address, unlike grpc-health-probe that often runs against
-  `127.0.0.1`. Be sure to configure your gRPC endpoint to listen on the Pod's IP address.
-- Built-in probes do not support any authentication parameters (like `-tls`).
+When using a gRPC probe, there are some technical details to be aware of:
+
+- The probes run against the pod IP address.
+  Be sure to configure your gRPC endpoint to listen on the Pod's IP address.
```
Comment (reylejano) [inline content/en/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md:280]: I think a hostname can be used but defaults to the pod's IP, according to the KEP
https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2727-grpc-probe
/sig node

Comment (sftim) [reply content/en/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md:280]: /lgtm cancel

**Comment by sftim** [comment]
@kubernetes/sig-node-pr-reviews does this look right?

File: content/en/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md (hunk 2 above)
Comment (tengqm) [reply content/en/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md:280]: Double checked. The endpoint has to listen on the Pod IP, but the probe can work using hostname or IP address.
Fixed the text for clarity.

**Comment by sftim** [comment]
(still) LGTM for docs
/lgtm

**Comment by sftim** [comment]
@kubernetes/sig-node-pr-reviews feel free to review this; it's optional but still useful

**Pull request merged**